src/modules/70-pipeline/hooks/useAddStepTemplate.ts

```diff
@@ -79,71 +77,52 @@ export function useAddStepTemplate(props: AddStepTemplate): AddStepTemplateRetur
-              if (pipelineStage && !pipelineStage.stage?.spec) {
-                set(pipelineStage, 'stage.spec', {})
-              }
-              if (pipelineStage && isNil(pipelineStage.stage?.spec?.execution)) {
-                if (event.isRollback) {
-                  set(pipelineStage, 'stage.spec.execution', { rollbackSteps: [] })
-                } else {
-                  set(pipelineStage, 'stage.spec.execution', { steps: [] })
-                }
-              }
-              executionRef?.stepGroupUpdated?.(newStepData.step)
-              addStepOrGroup(
-                event.entity,
-                pipelineStage?.stage?.spec?.execution as any,
-                newStepData,
-                event.isParallel,
-                event.isRollback
-              )
-              if (pipelineStage?.stage) {
-                await updateStage(pipelineStage?.stage)
-              }
-              if (!isCopied && template?.identifier && template?.childType) {
-                templateTypes[template.identifier] = template.childType
-                setTemplateTypes(templateTypes)
-              }
-              updateTemplateView({
-                isTemplateDrawerOpened: false,
-                templateDrawerData: {
-                  type: TemplateDrawerTypes.UseTemplate
-                }
-              })
-              updatePipelineView({
-                ...pipelineView,
-                isDrawerOpened: true,
-                drawerData: {
-                  type: DrawerTypes.StepConfig,
-                  data: {
-                    stepConfig: {
-                      node: newStepData.step,
-                      stepsMap: event.stepsMap,
-                      onUpdate: executionRef?.stepGroupUpdated,
-                      isStepGroup: false,
-                      addOrEdit: 'edit',
-                      hiddenAdvancedPanels: [AdvancedPanels.PreRequisites]
-                    }
-                  }
-                }
-              })
+  const addTemplate = async (event: ExecutionGraphAddStepEvent) => {
+    try {
+      const { template, isCopied } = await getTemplate({ templateType: 'Step', allChildTypes })
+      const newStepData = { step: createStepNodeFromTemplate(template, isCopied) }
+      const { stage: pipelineStage } = cloneDeep(getStageFromPipeline(selectedStageId || ''))
+      if (pipelineStage && !pipelineStage.stage?.spec) {
+        set(pipelineStage, 'stage.spec', {})
+      }
+      if (pipelineStage && isNil(pipelineStage.stage?.spec?.execution)) {
+        if (event.isRollback) {
+          set(pipelineStage, 'stage.spec.execution', { rollbackSteps: [] })
+        } else {
+          set(pipelineStage, 'stage.spec.execution', { steps: [] })
+        }
+      }
+      executionRef?.stepGroupUpdated?.(newStepData.step)
+      addStepOrGroup(
+        event.entity,
+        pipelineStage?.stage?.spec?.execution as any,
```

Comment:
Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232274)
```suggestion
                pipelineStage?.stage?.spec?.execution as unknown,
```

Reply:
This is an old code.. just moved it inside try catch.